From 884ff2fb0860dd98482dd1a552ca8dc15850df39 Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Mon, 7 Nov 2016 18:50:43 -0500
Subject: [PATCH 2087/2094] drm/ttm: Workaround memory allocation limits
 affecting KFD

Allow up to 15/16 of system memory to be used.

Don't enforce DMA32 limits. They don't matter for AMD GPUs that can
access at least 40 bits of physical address space.

Fix BUG: SWDEV-106528

Change-Id: I5d47bfa49459c62da49bbae6058c940b2d37b3f6
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/ttm/ttm_memory.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/ttm/ttm_memory.c b/drivers/gpu/drm/ttm/ttm_memory.c
index 29855be..cbb15be 100644
--- a/drivers/gpu/drm/ttm/ttm_memory.c
+++ b/drivers/gpu/drm/ttm/ttm_memory.c
@@ -257,9 +257,9 @@ static int ttm_mem_init_kernel_zone(struct ttm_mem_global *glob,
 
 	zone->name = "kernel";
 	zone->zone_mem = mem;
-	zone->max_mem = mem >> 1;
-	zone->emer_mem = (mem >> 1) + (mem >> 2);
-	zone->swap_limit = zone->max_mem - (mem >> 3);
+	zone->max_mem = mem - (mem >> 4); /* 15/16 */
+	zone->emer_mem = mem - (mem >> 4); /* 15/16 */
+	zone->swap_limit = zone->max_mem;
 	zone->used_mem = 0;
 	zone->glob = glob;
 	glob->zone_kernel = zone;
@@ -327,7 +327,8 @@ static int ttm_mem_init_dma32_zone(struct ttm_mem_global *glob,
 	 * No special dma32 zone needed.
 	 */
 
-	if (mem <= ((uint64_t) 1ULL << 32)) {
+	/*if (mem <= ((uint64_t) 1ULL << 32)) {*/
+	if (1) {
 		kfree(zone);
 		return 0;
 	}
-- 
2.7.4

